extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -282,24 +289,36 @@ StreamAppenderatorDriver newDriver(
     );
   }
 
-  KafkaConsumer<byte[], byte[]> newConsumer()
+  KafkaConsumer<byte[], byte[]> newConsumer() throws IOException
   {
     ClassLoader currCtxCl = Thread.currentThread().getContextClassLoader();
     try {
       Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
 
-      final Properties props = new Properties();
-
-      for (Map.Entry<String, String> entry : ioConfig.getConsumerProperties().entrySet()) {
-        props.setProperty(entry.getKey(), entry.getValue());
+      final Map<String, Object> configs = new HashMap<>();
```

Comment:
Since the json for consumerProperties would be as follows,
```
"consumerProperties": {
      "ssl.truststore.password": {
        "type": "foo",
        "passwordKey": "foo_key"
      }
    }
```
I used `Object` for it to get deserialized properly.

Reply:
I understand that , but in the code ,you are converting above blobs to string by converting them to PasswordProvider and then getPassword() on that. For other properties , we should still be getting strings, which can be casted to strings. so, not sure why this variable can't be `Map<String, String>` .

Reply:
Ah I misunderstood. Fixed